src/app/shared/services/share/share.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from "@angular/core";
+import { SyncServiceBase } from "../syncService.base";
+import { Share, ShareOptions } from "@capacitor/share";
+import { TemplateActionRegistry } from "../../components/template/services/instance/template-action.registry";
+
+@Injectable({
+  providedIn: "root",
+})
+export class ShareService extends SyncServiceBase {
+  constructor(private templateActionRegistry: TemplateActionRegistry) {
+    super("Share");
+    this.initialise();
+  }
+
+  private initialise() {
+    this.registerTemplateActionHandlers();
+  }
+
+  private registerTemplateActionHandlers() {
+    this.templateActionRegistry.register({
+      share: async ({ args }) => {
+        const [actionId, ...shareArgs] = args;
+        const childActions = {
+          text: async () => {
+            const [text] = shareArgs;
+            await this.share({ text: String(text) });
+          },
+          url: async () => {
+            const [url] = shareArgs;
+            await this.share({ url });
+          },
```

Comment:
nit(non-blocking) - Do we also plan to add support for sharing files? Might be useful to consider how to convert from the plh asset path to local file for sharing and including within the main methods

Reply:
Yes, that's the plan. It should be reasonably straightforward to use a method introduced by the remote assets work to handle that conversion, so I think it'd make sense to add the file sharing feature after #1923 is merged at least. I've updated #1951 accordingly.